src/core/friendly_errors/param_validator.js

```diff
@@ -68,45 +132,95 @@ function generateZodSchemasForFunc(func) {
     overloads = funcInfo.overloads;
   }
 
-  const createParamSchema = param => {
+  // Generate a schema for a single parameter that can be of multiple constants.
+  // Note that because we're ultimately interested in the value of the constant,
+  // mapping constants to their values via `constantsMap` is necessary.
+  //
+  // Also, z.num() can only be used for a fixed set of allowable string values.
+  // However, our constants sometimes have numeric or non-primitive values.
+  // Therefore, z.union() is used here instead.
+  const generateConstantSchema = constants => {
+    return z.union(constants.map(c => z.literal(constantsMap[c])));
+  }
+
+  // Generate a schema for a single parameter that can be of multiple
+  // non-constant types, i.e. `String|Number|Array`.
+  const generateUnionSchema = types => {
+    return z.union(types
+      .filter(t => {
+        if (!schemaMap[t]) {
+          console.log(`Warning: Zod schema not found for type '${t}'. Skip mapping`);
```

Comment:
Do you think we should make this throw an error so that it's obvious when we add a method that doesn't have a Zod schema yet?

Reply:
Yes absolutely! I used `console.log` right now because many objects are not supported yet. Added a TODO here for future reference!